_extras/yaml.md

```diff
@@ -0,0 +1,194 @@
+    type: boolean
+    inputBinding: # and this one too
+      position: 1
+      prefix: -f
+```
+
+The YAML above illustrates how you can build up complex nested object
+descriptions relatively quickly.
+The `inputs` map contains a single key, `example_flag`,
+which itself contains two keys, `type` and `inputBinding`,
+while one of these children, `inputBinding`,
+contains a further two key-value pairs (`position` and `prefix`).
+See the "Arrays" section below for more information about providing multiple
+values/key-value pairs for a single key.
+For comparison with the example YAML above,
+here is a graphical representation of the `inputs` object it describes.
+
+<div class="mermaid">
+graph TD
+  inputs --> example_flag
+  example_flag --> type
+  type --- bool((boolean))
+  example_flag --> inputBinding
+  inputBinding --> position
+  inputBinding --> prefix
+  position --- posval((1))
+  prefix --- prefval(('-f'))
+</div>
+
+### Arrays
+
+In certain circumstances it is necessary to provide
+multiple values or objects for a single key.
+As we've already seen in the "Mapped Objects" section above,
+more than one key-value pair can be mapped to a single key.
+However, it is also possible to define multiple values for a key
+without having to provide a unique key for each value.
+We can achieve this with an _array_,
+where each value is defined on its own line and preceded by `-`.
+For example:
+
+```yaml
+touchfiles:
+  - foo.txt
+  - bar.dat
+  - baz.txt
+```
+
+and a more complex example combining maps and arrays:
+
+```yaml
+exclusive_parameters:
+  type:
+    - type: record
+      name: itemC
+      fields:
+        itemC:
+          type: string
+          inputBinding:
+            prefix: -C
+    - type: record
+      name: itemD
+      fields:
+        itemD:
+          type: string
+          inputBinding:
+            prefix: -D
+```
+
+### JSON Style
+
+YAML is based on [JavaScript Object Notation (JSON)][json]
+and maps and arrays can also be defined in YAML using the native JSON syntax.
+For example:
+
+```yaml
+touchfiles: [foo.txt, bar.dat, baz.txt] # equivalent to first Arrays example
+```
+
+and:
+
+```yaml
+# equivalent to the `inputs` example in "Mapped Objects" above
+inputs: {example_flag: {type: boolean, inputBinding: {position: 1, prefix: -f}}}
+```
+
+Native JSON can be useful
+to indicate where a field is being left intentionally empty
+(such as `[]` for an empty array),
+and where it makes more sense
+for the values to be located on the same line
+(such as when providing option flags and their values in a shell command).
+However, as the second example above shows,
+it can severely affect the readability of a YAML file
+and should be used sparingly.
+
+### Reference
+
+The [Learn YAML in Y Minutes][yaml-y-mins] reference was very helpful for us
+while we wrote this guide.
+If you'd like to learn about some of the other features of YAML,
+which aren't valid for use with CWL,
+we highly recommend that you read it.
```

Comment:
I would remove this sentence. To the previous sentence I would add `, though it explains features of YAML that are not valid for CWL`